.travis.yml

```diff
@@ -23,16 +25,13 @@ install:
   - python setup.py install
   - python -m nltk.downloader wordnet
 
+env:
+  - TEST_SUITE=test/unit-test/unit_test.sh
+  - TEST_SUITE=test/functional-test/functional_test.sh
+  - TEST_SUITE=test/integration-test/integration_test.sh
+
 script:
-  - coverage run --source=. -m unittest discover -s test/functional-test
-  - coverage report -m
-  - coveralls
-  - coverage run --source=. -m unittest discover -s test/unit-test
-  - bash ext/doc_auto_deploy.sh
+  bash $TEST_SUITE
 
 after_success:
-  - coverage report -m
-  - codecov
-  - cd test
-  #- mprof clean && mprof run python -m unittest -v tests && cd ../profiler && python memory_profiler.py && echo "FOR UPDATED MEMORY CONSUMPTION, see image:" && curl -F "clbin=@memory_consumption.png" https://clbin.com
-  #- sfood ../lib/analysis ../lib/analysis/* ../ext ../lib/in_out/* ../lib/config.py ../lib/nickTracker.py ../lib/util.py ../lib/vis.py ../sample.py | sfood-graph | dot -Tpng -Gsize=9,15\! -Gdpi=300 -o call-graph-dependency.png && echo "FOR UPDATED DEPENDENCY, see image:" && curl -F "clbin=@call-graph-dependency.png" https://clbin.com
+  - bash ext/doc_auto_deploy.sh
```

Comment:
With the current structure, the doc_auto_deploy.sh is running in all the runs of the build matrix. A better solution is to mix the build stages with build matrix. Please see [example here](https://docs.travis-ci.com/user/build-stages/matrix-expansion/). The `bash ext/doc_auto_deploy.sh` command can be pushed to deploy stage. The default stage would be a test stage (this fact is also exploited in the example given).

Reply:
Sir, if we do the same as inside example and deploy is pushed to a separate stage, then a new VM is created and all the dependency installations are completed. i've pushed the commit already. you can have a view of the travis build. If this is fine, we can go ahead and merge but this deployment then got its's own 4.5 minute setup time thus pushing the total time to 15 mins.